src/autosuggest/virtual-list.tsx

```diff
@@ -26,22 +26,25 @@ const VirtualList = ({
   const [width, strutRef] = useContainerQuery(rect => rect.contentBoxWidth, []);
   useImperativeHandle(strutRef, () => scrollRef.current);
 
-  const rowVirtualizer = useVirtual({
-    items: autosuggestItemsState.items,
-    parentRef: scrollRef,
+  const rowVirtualizer = useVirtualizer({
+    count: autosuggestItemsState.items.length,
+    getScrollElement: () => scrollRef.current,
     // estimateSize is a dependency of measurements memo. We update it to force full recalculation
     // when the height of any option could have changed:
     // 1: because the component got resized (width property got updated)
     // 2: because the option changed its content (highlightText property controls the highlight and the visibility of hidden tags)
     // eslint-disable-next-line react-hooks/exhaustive-deps
     estimateSize: useCallback(() => 31, [width, highlightText]),
+    overscan: 5,
   });
 
+  const totalItems = autosuggestItemsState.items.length;
   useEffect(() => {
-    if (autosuggestItemsState.highlightType.moveFocus) {
-      rowVirtualizer.scrollToIndex(autosuggestItemsState.highlightedIndex);
+    const index = autosuggestItemsState.highlightedIndex;
+    if (autosuggestItemsState.highlightType.moveFocus && index >= 0 && index < totalItems) {
```

Comment:
What changed here? Why do we need to validate `index` now?

Reply:
In v3 there is an error thrown when executing scroll to an index for which there is no item defined, see https://github.com/TanStack/virtual/blob/main/packages/virtual-core/src/index.ts#L822

Reply:
ok for now, but doesn't it sounds like a problem with our code?

Reply:
I agree, there is an opportunity for improvement. In this particular case the issue comes from the fact we use -1 as the initial value of the highlighted index.